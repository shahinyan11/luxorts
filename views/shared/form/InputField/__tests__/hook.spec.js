import { renderHook } from '@testing-library/react-hooks';

import useContainer from '../hook';

describe('InputField useContainer hook', () => {
  const props = { name: 'name', numbersOnly: true };
  const { result } = renderHook(() => useContainer(props));

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  describe('onKeyDown method when numbersOnly equals true', () => {
    const event = { key: 'e', preventDefault: jest.fn() };

    it('shouldn`t calls `event.preventDefault()` when symbol is `Tab` ', () => {
      result.current.onKeyDown({
        ...event,
        keyCode: 9, // Tab
      });

      expect(event.preventDefault).not.toHaveBeenCalled();
    });

    it('should calls `event.preventDefault()` when symbol is not a number ', () => {
      result.current.onKeyDown(event);

      expect(event.preventDefault).toHaveBeenCalled();
    });
  });
});
