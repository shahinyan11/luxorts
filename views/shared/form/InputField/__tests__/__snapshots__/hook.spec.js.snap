// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField useContainer hook matches snapshot 1`] = `
Object {
  "field": Object {},
  "meta": Object {},
  "onKeyDown": [Function],
}
`;
